File: app/src/main/java/edu/gatech/donatrix/controllers/LoginActivity.java
package edu.gatech.donatrix.controllers;

import android.content.Intent;
import android.os.Bundle;
import android.os.StrictMode;
import android.support.v7.app.AppCompatActivity;
import android.view.Gravity;
import android.view.View;
import android.widget.EditText;
import android.widget.Toast;

import java.util.HashMap;
import java.util.Map;
import java.util.Objects;

import edu.gatech.donatrix.R;
import edu.gatech.donatrix.data.RESTCaller;
import edu.gatech.donatrix.model.User;

public class LoginActivity extends AppCompatActivity {

    private EditText emailField;
    private EditText passwordField;

    /**
     * Activity initializer
     *
     * @param savedInstanceState the activity state
     */
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_login);

        StrictMode.ThreadPolicy policy = new StrictMode.ThreadPolicy.Builder().permitAll().build();
        StrictMode.setThreadPolicy(policy);

        emailField = findViewById(R.id.loginEmailTextField);
        passwordField = findViewById(R.id.loginPasswordTextField);
    }

    /**
     * Called when login button is pressed
     *
     * @param view the current view
     */
    public void onLoginPressed(View view) {
        Map<String, Object> body = new HashMap<>();
        body.put("email", "" + emailField.getText());
        body.put("password", "" + passwordField.getText());

        Map<String, Object> response = RESTCaller.post(
                "https://donatrix-api.herokuapp.com/login", body);
        boolean success = User.testLoggedIn(body);
        Map<String, Object> user = (Map<String, Object>) response.get("user");

        if (success) {
            String type = (String) Objects.requireNonNull(user).get("type");
            if ("USER".equals(type)) {
                Intent intent = new Intent(LoginActivity.this, UserHomeActivity.class);
                startActivity(intent);
            } else if ("ADMIN".equals(type)) {
                Intent intent = new Intent(LoginActivity.this,
                        AdminHomeActivity.class);
                startActivity(intent);
            } else if ("LOCATION_EMPLOYEE".equals(type)) {
                Intent intent = new Intent(LoginActivity.this,
                        LocationEmployeeHomeActivity.class);
                intent.putExtra("location_id", (int) user.get("loc_id"));
                startActivity(intent);
            } else {
                Intent intent = new Intent(LoginActivity.this,
                        ManagerHomeActivity.class);
                startActivity(intent);
            }

        } else {
            Toast toast = Toast.makeText(this, "Invalid Login", Toast.LENGTH_SHORT);
            toast.setGravity(Gravity.CENTER, 0, 0);
            toast.show();
        }
    }
}

File: app/src/main/java/edu/gatech/donatrix/controllers/LoginActivity.kt
package edu.gatech.donatrix.controllers

import android.content.Intent
import android.os.Bundle
import android.os.StrictMode
import android.support.v7.app.AppCompatActivity
import android.view.Gravity
import android.view.View
import android.widget.EditText
import android.widget.Toast

import java.util.HashMap

import edu.gatech.donatrix.R
import edu.gatech.donatrix.data.RESTCaller
import edu.gatech.donatrix.model.User

class LoginActivity : AppCompatActivity() {

    private var emailField: EditText? = null
    private var passwordField: EditText? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_login)

        val policy = StrictMode.ThreadPolicy.Builder().permitAll().build()
        StrictMode.setThreadPolicy(policy)

        emailField = findViewById(R.id.loginEmailTextField)
        passwordField = findViewById(R.id.loginPasswordTextField)
    }

    fun onLoginPressed(view: View) {
        val body = HashMap<String, Any>()
        body["email"] = "" + emailField!!.text
        body["password"] = "" + passwordField!!.text

        val response = RESTCaller.post("https://donatrix-api.herokuapp.com/login", body)
        val success = User.testLoggedIn(body)
        val user = response["user"] as Map<String, Any>?

        if (success) {
            val type = user!!["type"] as String?
            if ("USER" == type) {
                val intent = Intent(this@LoginActivity, UserHomeActivity::class.java)
                startActivity(intent)
            } else if (type == "ADMIN") {
                val intent = Intent(this@LoginActivity, AdminHomeActivity::class.java)
                startActivity(intent)
            } else if (type == "LOCATION_EMPLOYEE") {
                val intent = Intent(this@LoginActivity, LocationEmployeeHomeActivity::class.java)
                intent.putExtra("location_id", user["loc_id"] as Int)
                startActivity(intent)
            } else {
                val intent = Intent(this@LoginActivity, ManagerHomeActivity::class.java)
                startActivity(intent)
            }

        } else {
            val toast = Toast.makeText(this, "Invalid Login", Toast.LENGTH_SHORT)
            toast.setGravity(Gravity.CENTER, 0, 0)
            toast.show()
        }
    }
}
